### PR TITLE
Rebase dangling XT patches onto RELEASE-4.10.0

### PR DESCRIPTION
**Pull request by aanisov**
Please pay your strict attention to patches named "FIXUP REBASE CONFLICTS" in order to recognize if the conflicts are resolved correctly.

One adopter patch is:

> [FIXUP: flask: adopt policy description changes](https://github.com/aanisov/xen/commit/d784863b33c5a4322ea3c64d713622b47b31f1e3)

Patches treated as odd and dropped out from rebase process:

> libxl: add PV display device driver interface
> xl: add PV display device commands
> docs: add virtual display driver information
> [1:1] xen/arm: allow to allocate 1/128/256/512 Mb memory chunks
> [1:1] HACK
> iommu/arm: ipmmu-vmsa: temp: Prevent IPMMU to be registered
> Revert "iommu/arm: ipmmu-vmsa: temp: Prevent IPMMU to be registered"
> Revert "[1:1] HACK"
> Revert "[1:1] xen/arm: allow to allocate 1/128/256/512 Mb memory chunks"
> Revert "docs: add virtual display driver information"
> Revert "xl: add PV display device commands"
> Revert "libxl: add PV display device driver interface"
> libxl: determine driver domain by libxl local entry
> Revert "libxl: determine driver domain by libxl local entry"

Patches already exist in XEN mainline(upstreamed):

> arm64:renesas: Introduce early console for Salvator-X board
> xen/arm: vgic: Check for vgic handler to be initialized before dereferencing it
> xen/arm: p2m: Check for p2m->domain to be initialized before releasing resources
> libxl: remove list callback from device framework
> libxl: provide typedefs for device framework functions
> libxl: use libxl__read_xenstore_mandatory in vtpm function
> libxl: use libxl__read_xenstore_mandatory in vdispl function
> libxl: fix disk listing function
> libxl: add back libxl_device_v{k,f}b_add
> libxl: remove unneeded DEVICE_ADD macro
> libxl: change vtpm to use generec add function
> libxl: fix memory leak in libxl__colo_save_setup
> libxl: change nic to use generec add function
> libxl: change disk to use generic getting list functions
> libxl: change vfb to use generec add function
> libxl: change vkb to use generec add function
> libxl: change p9 to use generec add function
> docs: add PV display driver information
> xl: add PV display device commands
> libxl: add vdispl device
> libxl: add generic functions to get and free device list
> libxl: add generic function to add device
> libxl, xl: change p9 to p9s
> tools/hotplug: create XEN_LOG_DIR at runtime

### CODE REVIEW

File: xen/arch/x86/mm.c [hunk 1]
```diff
@@ -2646,22 +2646,11 @@ static int __get_page_type(struct page_info *page, unsigned long type,
...
-=======
-                iommu_ret = iommu_unmap_pages(d,
-                                              mfn_to_gmfn(d, page_to_mfn(page)),
-                                              0);
-            else if ( type == PGT_writable_page )
-                iommu_ret = iommu_map_pages(d,
-                                            mfn_to_gmfn(d, page_to_mfn(page)),
-                                            page_to_mfn(page), 0,
-                                            IOMMUF_readable|IOMMUF_writable);
->>>>>>> d530dd9... iommu: Add extra order argument to the IOMMU APIs and platform callbacks
```
Comment (otyshchenko1) [inline xen/arch/x86/mm.c]: This is not correct. You should have left iommu_map_pages/iommu_unmap_pages

File: xen/arch/x86/mm/p2m.c [hunk 2]
```diff
@@ -1232,11 +1228,7 @@ int clear_identity_p2m_entry(struct domain *d, unsigned long gfn_l)
     {
         if ( !need_iommu(d) )
             return 0;
-<<<<<<< HEAD
-        return iommu_unmap_page(d, gfn_l);
-=======
-        return iommu_unmap_pages(d, gfn, 0);
->>>>>>> d530dd9... iommu: Add extra order argument to the IOMMU APIs and platform callbacks
+        return iommu_unmap_page(d, gfn_l, 0);
```
Comment (otyshchenko1) [inline xen/arch/x86/mm/p2m.c]: This is not correct as well. You should have left iommu_unmap_pages.
I am wondering have you ever tried to build this stuff under x86?

File: xen/arch/x86/mm/p2m.c [hunk 3]
```diff
@@ -1138,11 +1138,7 @@ int set_identity_p2m_entry(struct domain *d, unsigned long gfn_l,
     {
         if ( !need_iommu(d) )
             return 0;
-<<<<<<< HEAD
-        return iommu_map_page(d, gfn_l, gfn_l, IOMMUF_readable|IOMMUF_writable);
-=======
-        return iommu_map_pages(d, gfn, gfn, 0, IOMMUF_readable|IOMMUF_writable);
->>>>>>> d530dd9... iommu: Add extra order argument to the IOMMU APIs and platform callbacks
+        return iommu_map_page(d, gfn_l, gfn_l, 0, IOMMUF_readable|IOMMUF_writable);
```
Comment (otyshchenko1) [inline xen/arch/x86/mm/p2m.c]: This is not correct. You should have left iommu_map_pages.

File: xen/arch/x86/mm/p2m.c (hunk 2 above)
Comment (aanisov) [reply xen/arch/x86/mm/p2m.c]: No, I didn't.

**Comment by otyshchenko1** [comment]
BTW, do you plan to add here patches which are not in the tree?

**Pull request closed without merging**